scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2111,44 @@ def visit_call(self, node):
                     break
 
 
+class StringConcatenationChecker(checkers.BaseChecker):
+    """Custom pylint checker which prohibits use of string concatenation and
+     promotes string interpolation instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'string-concatenation'
+    priority = -1
+    msgs = {
+        'C0034': (
+            'at string %s, use string interpolation (\'string1%%s\' %% string2)'
+            ' rather than string concatenation (\'string1\' + \'string2\').',
+            'string-concatenation',
+            'Enforce use of string interpolation over string concatenation.',
+        ),
+    }
+
+    def visit_binop(self, node):
+        """Called for every '+' operator to prohibit usage of string
+         concatenation. It covers only those cases where at least one of the
+         operands is astroid.nodes.Const.
+
+        Args:
+            node: astroid.node.BinOp. Node to access module content.
+        """
+        if node.op == b'+':
+            for operand in node.get_children():
+                if (isinstance(operand, astroid.nodes.Const) and
+                        ('str' in operand.pytype() or
+                         'unicode' in operand.pytype())):
```

Comment:
Maybe convert this into a helper function with a meaningful name that makes it clear what is it that we are exactly doing here. (I understand it, but for the sake of readability it's a good practice to convert complicated conditions into a helper function)

Reply:
Thanks for the suggestion Done!!

Reply:
Thanks for the suggestion, implemented it.